whipper/image/table.py

```diff
@@ -443,30 +408,37 @@ def _getMusicBrainzValues(self):
         # number of first track
         result.append(1)
 
-        # number of last audio track
-        result.append(self.getAudioTracks())
+        # number of last audio track (default: number of audio tracks)
+        lastTrack = self.getAudioTracks()
+        result.append(lastTrack)
 
-        leadout = self.leadout
-        # if the disc is multi-session, last track is the data track,
-        # and we should subtract 11250 + 150 from the last track's offset
-        # for the leadout
-        if self.hasDataTracks():
-            assert not self.tracks[-1].audio
-            leadout = self.tracks[-1].getIndex(1).absolute - 11250 - 150
-
-        # treat leadout offset as track 0 offset
-        result.append(150 + leadout)
+        dataTrackLast = False
+        additional = 0
+        offsets = []
 
         # offsets of tracks
-        for i in range(1, 100):
-            try:
-                track = self.tracks[i - 1]
-                if not track.audio:
-                    continue
-                offset = track.getIndex(1).absolute + 150
-                result.append(offset)
-            except IndexError:
-                pass
+        for i in range(0, len(self.tracks)):
+            track = self.tracks[i]
+            if not track.audio:
+                # if the data track is not at the end
+                if i < len(self.tracks) - 1:
+                    additional += 1
+                else:
+                    # if the data track is last
+                    dataTrackLast = True
+                    sectors = self.tracks[-1].getIndex(1).absolute - 11400
```

Comment:
@MerlijnWajer , so you mean this line `11400` should be `11250 - 150` ?

Reply:
Not necessarily, I just don't recall right now where the numbers comes from. Does discid or the musicbrainz library not have a way to generate a submit URL? cc @Freso

Reply:
> Does discid or the musicbrainz library not have a way to generate a submit URL?

The class `discid.Disc` (from the `discid` module) has a `submission_url` object which should be just that: [link](https://python-discid.readthedocs.io/en/latest/api/#discid.Disc.submission_url).

Reply:
@ABCbum 11250 - 150 ≠ 11400 (11100). 11400 = 11250 + 150.
150 frames is the "standard" 2 seconds pre-gap. 11250 is 150 seconds. Don't know if those values are constant in this case (I think that would be strange)...

Reply:
@JoeLametta , I think what i meant was `-11250 -150` 😅😅. But anyway, that piece of code i tried to keep it [original](https://github.com/whipper-team/whipper/blob/8f5559ebc8/whipper/image/table.py#L449#L458), how that works is a little weird after i test with tons of commit earlier. 

From what i see, the gap belongs to the last audio track if the data track is originally at last place, like [Lady Hawker test case](https://github.com/whipper-team/whipper/blob/8f5559ebc8/whipper/test/test_image_toc.py#L200). However, if the data track is 'merged' or appended, meaning that the original TOC ended with out the gap which is in [this test case](https://github.com/whipper-team/whipper/blob/8f5559ebc8/whipper/test/test_image_toc.py#L236), after merging a data track, that gap belongs to neither data track nor last audio track.
To illustrate: 
- Not appended: `audio audio ... (audio + gap) dataTrackStarts`.
- Appended: `audio audio ... audio + gap + dataTrackStarts(appended data track)`.

That's why i couldn't use `getTrackEnd(lastTrack)` with discs(or tocs) that are merged (with an additional data track at the end and the origonal TOC doesn't have gap). This means getting the `dataTrackStarts - gap` should be the correct last audio ending sector in both cases.

Hopefully what i see is right and i was able to explain it to you guys.

Reply:
@MerlijnWajer I've noticed that class also has a [`freedb_id`](https://python-discid.readthedocs.io/en/latest/api/#discid.Disc.freedb_id) object: do you think it could replace `whipper/whipper/extern/freedb.py`?

Reply:
I've added the ending line from its commit too.